spec/models/user_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+
+describe User do
+  it "validates for an incorrect password length" do
+    expect(FactoryGirl.build(:user, password: "abc123")).to_not be_valid
+  end
+
+  it "validates for an correct password length" do
+    user = FactoryGirl.create(:user)
+
+    expect(user).to be_valid
+  end
+
+  it "validates for a non-unique username" do
+    FactoryGirl.create(:user)
+    expect(FactoryGirl.build(:user, username: "SayWhaaaaatttt5")).to_not be_valid
+  end
+
+  it "validates for a unique username" do
+    user = FactoryGirl.create(:user)
+
+    expect(user).to be_valid
+  end
+
+  it "validates for a non-unique email" do
+    FactoryGirl.create(:user)
+    expect(FactoryGirl.build(:user, email: "faketestuser8@example.com")).to_not be_valid
```

Comment:
So, there is a problem with this test. If I clear our the test database, faketestuser8@example.com won't exist, and this test will fail. I think what you want to do is this:

```
user = FactoryGirl.create(:user)
expect(FactoryGirl.build(:user, email: user.email)).to_not be_valid
```